spinalcordtoolbox/scripts/sct_image.py

```diff
@@ -355,8 +356,22 @@ def main(argv: Sequence[str]):
         printv(im_in.absolutepath)
         im_out = [change_orientation(im_in, arguments.setorient)]
 
-    elif arguments.setorient_data is not None:
-        im_out = [change_orientation(im_in, arguments.setorient_data, data_only=True)]
+    elif arguments.flip is not None:
+        if dim_list.index(arguments.flip) >= len(im_in.data.shape):
+            parser.error(f"Cannot flip axis '{arguments.flip}' on an image with n_dim = {len(im_in.data.shape)}.")
+        im_out = im_in.copy()
+        im_out.data = np.flip(im_out.data, axis=dim_list.index(arguments.flip))
+        im_out = [im_out]
+
+    elif arguments.transpose is not None:
+        if len(arguments.transpose) != len(im_in.data.shape):
+            parser.error(f"Transpose argument '{','.join(arguments.transpose)}' must match number of image dimensions "
+                         f"in the input image ({len(im_in.data.shape)}).")
+        im_out = im_in.copy()
+        transpose_numerical = [dim_list.index(axis)  # Convert [x, y, z, t] to [0, 1, 2, 3]
+                               for axis in arguments.transpose]
+        im_out.data = np.transpose(im_out.data, axes=transpose_numerical)
```

Comment:
I'll note that numpy has decent error messages for some of the possible errors in the axis list:
```py3
ValueError: repeated axis in transpose
numpy.AxisError: axis 3 is out of bounds for array of dimension 2
ValueError: axes don't match array
```

Reply:
Just to clarify, is this a suggestion to allow numpy to handle the errors instead of trying to catch them ourselves?

```python
sct_image -i t2.nii.gz -transpose z,x,x -o repeated_axes.nii.gz
# Before/After (this case wasn't covered by my error message)
"ValueError: repeated axis in transpose"

sct_image -i t2.nii.gz -transpose z,y,x,t -o too_many_axes.nii.gz
# Before (ours)
"sct_image: error: Transpose argument 'z,y,x,t' must match number of image dimensions in the input image (3)."
# After (numpy)
"ValueError: axes don't match array"

sct_image -i t2.nii.gz -flip t -o 3d_doesnt_have_t_dim.nii.gz
# Before (ours)
"sct_image: error: Cannot flip axis 't' on an image with n_dim = 3."
# After (numpy)
"numpy.AxisError: axis 3 is out of bounds for array of dimension 3"
```

I agree that numpy basically covers the cases that we'd worry about, and that my messages aren't super necessary. The only issues I have are:

- The arguments are currently specified in terms of axes labels (x,y,z,t) rather than 0-based indexing (0,1,2,3), so while numpy's error messages are informative for devs, they're might be a bit confusing for users, due to the mention of e.g. "axis 3"
- Numpy's error messages are also a bit terse in general?
- They also spit out a bit of an ugly traceback, with `_wrapfunc` and `bound` functions in the stack trace. 

I'm fine with removing my added input checking, but am curious about what your thoughts are, too. :)

Reply:
No no, your existing messages are fine: they prevent the ugly/confusing traceback, and expressing things in terms of (x, y, z, t) is nice.

But they don't catch every possible user error, for example:
```
sct_image -i t2.nii.gz -transpose z,x,x -o repeated_axes.nii.gz
sct_image -i t2.nii,gz -transpose x,y,t -o 3d_with_no_t.nii.gz
```
I was just noting that, even in these non-covered cases, the user will get a reasonable error message (directly from numpy).

Reply:
If you just wanted to _check_ that the axes are valid for transposition, I think this bit of code would do it exactly:
```py3
sorted(arguments.transpose) == sorted(DIM_LIST[:len(im_in.data.shape)])
```
But if you want to output sensible messages for different kinds of errors (wrong number of axes, repeated axis, out-of-range axis, invalid letter for an axis), that would be a bunch of code.

I think the most common source of errors is most likely to be wrong number of axes, which your code handles, so that should be fine.